### PR TITLE
Conversation handling

### PR DESCRIPTION
**Pull request by stephen-lt-sl**
@stevenhillcox @sievins @jebbinSCL

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -13,9 +13,14 @@
             <h2>List of registered users</h2>
             <ul>
                 <li ng-repeat="user in users">
-                    {{user.name}}
+                    {{user.name}} - <a href="" ng-click="getConversation(user.id)">(Chat)</a>
                 </li>
             </ul>
+            <div>
+                <div ng-repeat="conversation in conversations" style="width:200px; height:400px; border:2px solid blue; float: left">
+
```
Comment (jebbinSCL) [inline public/index.html:21]: I assume this is a placeholder for the next pull request. If you do this, make sure you add a comment or simple element that makes it obvious that this is a placeholder.

File: public/js/main.js [hunk 2]
```diff
@@ -15,5 +19,43 @@
                 $scope.loginUri = result.data.uri;
             });
         });
+
+        function addConversation(newConversation) {
```
Comment (jebbinSCL) [inline public/js/main.js:23]: What is addConversation and getConversation meant to be doing? There are a few things about them that seem odd. First, it isn't clear why a function called getConversation should call addConversation. Also, it isn't clear why getConversation contains a post request, and why it contains logging for updating or adding a conversation. Is getConversation correctly named?

File: server/server.js [hunk 3]
```diff
@@ -86,5 +89,75 @@ module.exports = function(port, db, githubAuthoriser) {
...
+                    var conversationData = {};
+                    for (var key in conversation) {
+                        if (key === "_id") {
+                            conversationData.id = conversation._id;
+                        } else {
+                            conversationData[key] = conversation[key];
+                        }
+                    }
+                    res.json(conversationData);
+                }
```
Comment (jebbinSCL) [inline server/server.js:115]: Bracketing is incorrect here.

**Comment by stephen-lt-sl** [comment]
Just closing this now because I've spent some time rebasing all the work done in the user-chat branch so that it can be placed into several independent feature branches, which should make for a much cleaner history - as well as greatly simplified review

**Pull request closed without merging**